lib/file-monitor.c

```diff
@@ -168,7 +169,7 @@ static void _inotify_event_handler(void *c, struct inotify_event *event)
 {
   FileMonitor *self = (FileMonitor *) c;
 
-  if (g_strcmp0(self->file_name, event->name) == 0)
+  if (g_strcmp0(self->file_basename, event->name) == 0)
```

Comment:
shouldn't we accept both the basename and the filename? do all event types name the directory we subscribed for?

Reply:
In this unit, it is always the file name that inotify returns.

Reply:
Sorry, I didn't catch that we watch for the dirname and this is one is the basename.  Also it looks like the iv wrapper takes care of checking the dirname (event->wd) when dispatching the callback. But I couldn't find out if watching multiple files in the same directory would work correctly. It might also be worth checking in an e2e test if atomic writes by rename()ing a temporary file to the final place is handled correctly (it should be based on a quick look at the code).